packages/fast-usdc/src/exos/advancer.js

```diff
@@ -152,12 +154,22 @@ export const prepareAdvancerKit = (
               return;
             }
 
+            const { zcfSeat: tmpSeat } = zcf.makeEmptySeatKit();
             try {
-              const payment = await assetManagerFacet.borrow(advanceAmount);
-              const depositV = E(poolAccount).deposit(payment);
+              const amountKWR = harden({ USDC: advanceAmount });
+              borrowerFacet.borrow(tmpSeat, amountKWR);
+
+              const depositV = zoeTools.localTransfer(
```

Comment:
the comment below says that `depositHandler` handles failures in `localTransfer`, so it seems odd to have this inside the `try`

I suppose we can think again when we do #10510 (comprehensive error testing) but I struggle to be confident that we will.

Reply:
> we safely assume we'll be in the same turn?

yes, it's statically evident that there are no turn boundaries between `.getBalance()` and the return from `.borrow()`.